### PR TITLE
Merge client & server WIP

### PR DESCRIPTION
**Pull request by brech1**
# Description

Solves #186

The aim of this PR is to simplify the protocol and merge the client and server crates.

The proof generation will be embedded in the client CLI.

### CODE REVIEW

**Comment by lazovicff** [comment]
Is it possible for you to switch to main repo, instead of working from fork.
For now, you can create a branch on main repo and merge this code into it.

File: client/src/manager.rs [hunk 1]
```diff
@@ -208,32 +225,19 @@ impl Manager {
 		// --- END ---
```
Comment (lazovicff) [inline client/src/manager.rs:225]: Maybe we should move this into the test `should_calculate_proof`, instead of doing sanity checks

Comment (brech1) [reply client/src/manager.rs:214]: Sure! Makes sense, the snippet only runs on unoptimized builds

**Comment by brech1** [comment]
> Is it possible for you to switch to main repo, instead of working from fork? For now, you can create a branch on main repo and merge this code into it.

Yes, I just pushed some updates and we can merge this to the new branch on the main repo.

**Pull request merged**